src/utilities/string.ts

```diff
@@ -11,6 +11,8 @@ export function toTitle(string: string) {
     .replace(/-/g, ' ');
 }
 
+const isUrlSafe = (input: string) => /^[\w.~-]*$/.test(input);
```

Comment:
shall we just define it inline since it's very short?

Reply:
did, PTAL.